src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java

```diff
@@ -125,7 +125,7 @@ public void init() {
      * Destroy the check. It is being retired from service.
      */
     public void destroy() {
-        // No code by default, should be overridden only by demand at subclasses
+        context.remove();
```

Comment:
Isn't this considered a breaking change? Before this change, you could override `destroy` in an implementing class and not need to call the super class.

Look at our own implementation checks like `RegexpOnFilename`.
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpOnFilenameCheck.java#L247
We override `init` method but never call super init.

Reply:
this is not a breaking change ... bug as maximum, and even misc (as almost nobody notice any difference). We are not MT tool, this change is literally to make CI happy.

Reply:
Then this will make things weirder when we do become a MT tool. Implementation of MT will not be marked as breaking compatibility but a new feature and then we are expecting users to just know that they should be calling the super destroy. It is going to be up to us now to remember to recheck all checks for this when we get to it.

Reply:
it is ok, eventually it become clear and all Checks become fixed. This should not block us now.

Reply:
I still think this makes it a breaking change, as well as bug.

Reply:
we can put on it breaking label if you wish .... I just try to not attract that much attention to this code change for plugin maintainers.

Reply:
If we declare this as a breaking change, then we need to determine which methods we would also do this for. As showed above for `init`, we don't call super for it. If we break for `destroy`, I feel we should break for all of them to make it easier to remember. I would think we would want to make this clear in xdoc/javadoc that they should be calling the super if they ever override the class.

Reply:
> I just try to not attract that much attention to this code change for plugin maintainers

How about we ask a plugin maintainer if they consider this a breaking change?

Reply:
I do not know what for such methods .... I do not want to explain user how to use them. We will come back to this ones we start finalizing our API.

> if they consider this a breaking change?

I do not know such plugins. I would make this issue as "misc".

Reply:
Is it possible to make a check to verify that either this method is not overridden, or it is overridden and calls the base class method of the same name?

Reply:
We can to something like this https://stackoverflow.com/a/34180574/1015848 in separate issue.